package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -1,246 +1,266 @@
+        - A task node is added as a child to a modular pipeline if it has the namespace of that modular pipeline
+        - Inputs/Outputs of a task node are added as children to the modular pipeline if they are not inputs/outputs of that modular pipeline
 
-    def add_child(self, modular_pipeline_id: str, child: ModularPipelineChild):
-        """Add a child to a modular pipeline.
         Args:
-            modular_pipeline_id: ID of the modular pipeline to add the child to.
-            child: The child to add to the modular pipeline.
-        Example:
-            >>> modular_pipelines = ModularPipelinesRepository()
-            >>> modular_pipeline_child = ModularPipelineChild(
-            ...     id="dataset",
-            ...     type=GraphNodeType.DATA
-            ... )
-            >>> modular_pipelines.add_child("data_science", modular_pipeline_child)
-            >>> data_science_pipeline = modular_pipelines.get_or_create_modular_pipeline(
-            ...     "data_science"
-            ... )
-            >>> assert data_science_pipeline.children == {modular_pipeline_child}
+            modular_pipeline_id (str): The ID of the modular pipeline to add children to.
+            task_nodes (List[GraphNode]): The task nodes (and it's related datasets/parameters) to add as children.
         """
         modular_pipeline = self.get_or_create_modular_pipeline(modular_pipeline_id)
-        modular_pipeline.children.add(child)
-
-    def extract_from_node(self, node: GraphNode) -> Optional[str]:
-        """Extract the namespace from a graph node and add it as a modular pipeline node
-        to the modular pipeline repository.
+        all_inputs_outputs = set(modular_pipeline.inputs) | set(
+            modular_pipeline.outputs
+        )
 
-        Args:
-            node: The GraphNode from which to extract modular pipeline.
-        Returns:
-            ID of the modular pipeline node added to the modular pipeline repository if found.
-        Example:
-            >>> modular_pipelines = ModularPipelinesRepository()
-            >>> model_output_node = GraphNode.create_data_node(
-            ...     "data_science.model_output", layer=None, tags=set(), dataset=None
-            ... )
-            >>> modular_pipelines.extract_from_node(model_output_node)
-            'data_science'
-            >>> assert modular_pipelines.has_modular_pipeline("data_science")
-        """
-
-        # There is no need to extract modular pipeline from parameters
-        # because all valid modular pipelines are encoded in either a TaskNode or DataNode.
-        if isinstance(node, ParametersNode):
-            return None
-
-        modular_pipeline_id = node.namespace
-        if not modular_pipeline_id:
-            return None
+        parent_modular_pipeline_id = (
+            modular_pipeline_id.split(".")[0] if "." in modular_pipeline_id else None

```

Comment:
so for nested modular pipeline `a.b.c` , the parent would be `a` and not `a.b` ? I would like to understand this, lets connect :)